src/Servers/Connections.Abstractions/src/Features/ITlsHandshakeFeature.cs

```diff
@@ -3,6 +3,10 @@
 
 using System.Security.Authentication;
 
+#if NET7_0_OR_GREATER
```

Comment:
That was copy-pasted.  Do you want me to update `TlsConnectionCallbackContext` and `TlsConnectionCallbackOptions` too?

Reply:
Yes please

Reply:
I actually like using the versioned defines, it's self-documenting for when the feature was added.
And this one would actually be NET8_0_OR_GREATER.

Reply:
I'm getting some mixed messages here. 😆

Reply:
On the one hand, @BrennanConroy is actually on the team that owns this code.  On the other hand, that seems like a really flaky way to capture introduction versions, especially since the type has no actual dependency on .net 8.  I'm leaning `NETCOREAPP` unless someone feels strongly.

Reply:
Git blame is better for code history 😁 
NET8_0 implies there's something specific required from .NET 8 that wasn't available previously. I'll check if I can find any other examples, but I think we've used the more general defines elsewhere.

Reply:
We're inconsistent at best 😆 
https://github.com/dotnet/aspnetcore/search?q=NETCOREAPP
https://github.com/dotnet/aspnetcore/search?q=NET7_0_OR_GREATER

I'd still suggest the most general option.

Reply:
We kind of do look consistent actually. We use versioned defines for things that actually require the version or higher, and non-versioned for things that need more than netstandard2.0/2.1. Which actually goes against what I originally said here. DIMs work in netcoreapp so that's probably the define that should be used.